### PR TITLE
Update region format

### PR DESCRIPTION
**Pull request by yux0**
## What was changed
Add region format validation

## Why?
Add region format validation

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: app/namespace.go [hunk 1]
```diff
@@ -288,19 +288,18 @@ func (c *NamespaceClient) addRegion(ctx *cli.Context) error {
...
-	if len(cloudProvider) == 0 {
-		return fmt.Errorf("namespace cloud provider is required")
+	if err := utils.ValidateCloudProviderAndRegion(regionString); err != nil {
+		return err
 	}
 
 	res, err := c.cloudAPIClient.AddNamespaceRegion(c.ctx, &cloudservice.AddNamespaceRegionRequest{
 		Namespace:        ctx.String(NamespaceFlagName),
-		Region:           fmt.Sprintf("%s-%s", cloudProvider, region),
+		Region:           regionString,
```
Comment (laniehei) [inline app/namespace.go:302]: can we do "%s:%s" like in the other cli?

File: app/namespace.go [hunk 2]
```diff
@@ -683,15 +681,10 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
 				},
 				&cli.StringFlag{
 					Name:     namespaceRegionFlagName,
-					Usage:    "New region to add to the namespace.",
+					Usage:    "New region to add to the namespace. Please use format <cloud provider>-<region name>.",
```
Comment (laniehei) [inline app/namespace.go:684]:
```suggestion
					Usage:    "New region to add to the namespace. Please use format <cloud provider>:<region name>.",
```

File: app/namespace.go [hunk 3]
```diff
@@ -1393,15 +1386,10 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
 				},
 				&cli.StringFlag{
 					Name:     namespaceRegionFlagName,
-					Usage:    "The region to failover to",
+					Usage:    "The new region to failover. Please use format <cloud provider>-<region name>.",
```
Comment (laniehei) [inline app/namespace.go:1389]:
```suggestion
					Usage:    "The new region to failover. Please use format <cloud provider>:<region name>.",
```

Comment (yux0) [reply app/namespace.go]: This is the new cloud api format. I cannot change it.

File: app/namespace.go (hunk 2 above)
Comment (yux0) [reply app/namespace.go]: This is the new cloud api format. I cannot change it.

File: app/namespace.go [hunk 4]
```diff
@@ -298,9 +299,14 @@ func (c *NamespaceClient) addRegion(ctx *cli.Context) error {
 		return fmt.Errorf("namespace cloud provider is required")
 	}
 
+	targetRegion := fmt.Sprintf("%s-%s", cloudProvider, region)
```
Comment (jumbogo) [inline app/namespace.go:302]: Do we already have a cloud provider flag for tcld?

Comment (yux0) [reply app/namespace.go:302]: yes. I added a optional one.

**Review by laniehei — APPROVED** [review]

**Pull request merged**